Assets/Scripts/SS3D/Systems/Gamemodes/GamemodeObjective.cs

```diff
@@ -15,12 +15,62 @@ public class GamemodeObjective : ScriptableObject, IGamemodeObjective
         public event Action<GamemodeObjective> OnGamemodeObjectiveUpdated;
 
         private int _id;
-        private string _title;
+        [SerializeField] private string _title;
+        [SerializeField] private CollaborationType _collaborationType;
+        [SerializeField] private Alignment _alignmentRequirement;
+        [SerializeField] private int _minAssignees = 1;
+        [SerializeField] private int _maxAssignees = 1;
         private ObjectiveStatus _status;
         private string _assigneeCkey;
 
         [SerializeField] public List<IEvent> ListensToEvent;
 
+        /// <summary>
+        /// No-arg constructor
+        /// </summary>
+        public GamemodeObjective()
+        {
+        }
+
+        /// <summary>
+        /// Constructor allowing creation of specific objective data.
+        /// </summary>
+        /// <param name="title">Title of the objective (visible to player)</param>
+        /// <param name="collaborationType">Whether the objective is individual, competitive or cooperative</param>
+        /// <param name="alignment">Whether the objective is valid for antagonists, non-antagonists or both</param>
+        /// <param name="minAssignees">Minimum number of players required for this objective</param>
+        /// <param name="maxAssignees">Maximum number of players required for this objective</param>
+        public GamemodeObjective(string title, CollaborationType collaborationType, Alignment alignment, int minAssignees, int maxAssignees)
+        {
+            this._title = title;
```

Comment:
remove all this.variable, if the variable is private, it'll have the _ on it, so there's no reason to use this.variable.